charms/worker/k8s/src/charm.py

```diff
@@ -269,6 +269,7 @@ def _configure_datastore(self, config: BootstrapConfig):
                 datastore,
                 ", ".join(SUPPORTED_DATASTORES),
             )
+            status.add(ops.BlockedStatus(f"Invalid datastore: {datastore}"))
```

Comment:
if you don't raise an error here... the reconciler will continue -- deploying with the "dqlite" datastore